parser.py

```diff
@@ -437,17 +437,18 @@ def write(self, fname):
 class PDDLProblemParser(PDDLParser):
     """PDDL problem parsing class.
     """
-    def __init__(self, problem_fname, domain_name, types, predicates):
+    def __init__(self, problem_fname, domain_name, types, predicates, action_names):
         self.problem_fname = problem_fname
         self.domain_name = domain_name
         self.types = types
         self.predicates = predicates
+        self.action_names = action_names
         self.uses_typing = not ("default" in self.types)
 
         self.problem_name = None
         # Set of objects, each is a structs.TypedEntity object.
         self.objects = None
-        # Set of fluents in initial state, each is a structs.Literal.
+        # Set of fluents in initial state, each is a structs.State.
```

Comment:
Do we really want to propagate this change into the `PDDLProblemParser`? This is genuine question, not a rhetorical one! One one hand, I can see an advantage to having a single concept of "state" shared across all classes, but, on the other, it might be best to keep all the raw/low-level PDDL stuff as standard and restrict the "fancy" state to `PDDLEnv` stuff. What do you think?

Reply:
Hmm yeah good question... I think keeping parsing "raw" is a good idea. I removed States from parser.py.